static/src/javascripts-legacy/projects/common/modules/experiments/tests/tailor-survey.js

```diff
@@ -0,0 +1,152 @@
+define([
+    'bean',
+    'fastdom',
+    'Promise',
+    'lodash/functions/debounce',
+    'common/utils/config',
+    'common/utils/cookies',
+    'common/utils/storage',
+    'common/utils/mediator',
+    'common/utils/fastdom-promise',
+    'common/utils/private-browsing',
+    'text!common/views/experiments/tailor-survey.html',
+    'common/utils/fetch-json',
+    'lodash/collections/forEach'
+], function (
+    bean,
+    fastdom,
+    Promise,
+    debounce,
+    config,
+    cookies,
+    storage,
+    mediator,
+    fastdomPromise,
+    privateBrowsing,
+    quickSurvey,
+    fetchJson,
+    forEach
+) {
+    return function () {
+        this.id = 'TailorSurvey';
+        this.start = '2017-01-25';
+        this.expiry = '2017-03-31';
+        this.author = 'Manlio';
+        this.description = 'Testing Tailor surveys';
+        this.audience = 0.01;
+        this.audienceOffset = 0.7;
+        this.successMeasure = 'We can show a survey on Frontend to regular users only (as decided by Tailor)';
+        this.audienceCriteria = 'All users';
+        this.dataLinkNames = 'Tailor survey';
+        this.idealOutcome = '';
+
+        this.canRun = function () {
+            return !(config.page.isAdvertisementFeature) &&
+                config.page.contentType === 'Article'
+        };
+
+        function callTailor(bwid) {
+            var endpoint = 'https://tailor.guardianapis.com/suggestions?browserId=' + bwid;
+            return fetchJson(endpoint, {
+                type: 'json',
+                method: 'get'
+            });
+        }
+
+        function renderQuickSurvey() {
+            var bwid = cookies.get('bwid');
+            var hasSeenTheSurveyAlready = cookies.get('GU_TAILOR_SURVEY_1') || false;
+
+            if (bwid && !hasSeenTheSurveyAlready) {
+                return callTailor(bwid).then(function (response) {
+                    return response.userDataForClient.regular;
+                }).then(function(isRegular) {
+                    if (isRegular) {
+                        cookies.add('GU_TAILOR_SURVEY_1', 1, 100); // do not show this survey to the user for the next 100 days
+
+                        return fastdomPromise.write(function () {
+                            var article = document.getElementsByClassName('content__article-body')[0];
+                            var insertionPoint = article.getElementsByTagName('p')[1];
+                            var surveyDiv = document.createElement('div');
+                            surveyDiv.innerHTML = quickSurvey;
+                            article.insertBefore(surveyDiv, insertionPoint);
+                        });
+                    }
+                });
+            }
+        }
+
+        function disableRadioButtons(buttonClassName) {
+            var radioButtons = document.getElementsByClassName(buttonClassName);
+            for (var i = 0; i < radioButtons.length; i++) {
```

Comment:
... of course pulling in an external library is simpler than writing a for loop? :trollface: